### ISSUE TITLE
New Port To Python

### ISSUE THREAD

**Issue by Helw150** [lopuhin/python-adagram#13]
Hey @lopuhin! I am interested in starting a new active development to port AdaGram to PyTorch. 

Would you mind giving me a run down on the issues you faced with your port's training steps? Were the issues purely to do with multiple workers (as the previous edition of the README said) or were there deeper issues that make your port not a valid model to base my port from?

**Comment by lopuhin** [comment]
Hey @Helw150 that's great! At first I though that the issue was due to multiple workers, because single worker seemed to produce slightly better results on smaller datasets. But even a single worker didn't produce expected results on the full dataset. So I think there must be some issue in how I ported the C code from Julia, or in how I re-wrote some bits of Julia into C to keep performance. Note that Julia is using 1-based indexing :) So due to above, I suggest starting from the original source, not from my port. But I'm confident that inference produces the same results as in the original.
